drake_ros_core/src/ros_subscriber_system.cpp

```diff
@@ -67,6 +67,7 @@ RosSubscriberSystem::RosSubscriberSystem(
     std::bind(&RosSubscriberSystemPrivate::handle_message, impl_.get(), std::placeholders::_1));
 
   DeclareAbstractOutputPort(
+    topic_name,
```

Comment:
[`LcmSubscriberSystem` uses `drake::systems::kUseDefaultName` here](https://github.com/RobotLocomotion/drake/blob/1df1c47241ae0c9ce1f45198cbb6ce0abcbb1269/systems/lcm/lcm_subscriber_system.cc#L50). I'd recommend using that instead.

It's also using a new method `DeclareStateOutputPort` as of https://github.com/RobotLocomotion/drake/pull/15563 which says it avoids downstream systems from unnecessarily recalculating stuff. Maybe that'd be a good thing to make `RosSubscriberSystem` use in a follow up PR.

Reply:
See #12 for an alternative to this change

Reply:
Those are both excellent points. Thanks for the PR!

Reply:
I would prefer to keep this library and close to in sync with `LcmSubscriberSystem` as possible.